Lib/test/test_imp.py

```diff
@@ -378,6 +378,40 @@ def test_find_and_load_checked_pyc(self):
             mod = imp.load_module('mymod', file, path, description)
         self.assertEqual(mod.x, 42)
 
+    def test_issue98354(self):
+        # _imp.create_builtin should raise TypeError
+        # if 'name' attribute of 'spec' argument is not a 'str' instance
```

Comment:
This test fails (crashes) if you remove the import.c change, right?

Reply:
Yes, it crashes if i revert changes in import.c.